contracts/erc7683/ERC7683Depositor.sol

```diff
@@ -0,0 +1,293 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./Permit2OrderLib.sol";
+import "../external/interfaces/IPermit2.sol";
+import "../interfaces/V3SpokePoolInterface.sol";
+
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./Permit2OrderLib.sol";
+
+/// @title CrossChainOrder type
+/// @notice Standard order struct to be signed by swappers, disseminated to fillers, and submitted to settlement contracts
+struct CrossChainOrder {
```

Comment:
nit: might make sense to move the canonical ERC7683 types into a separate file, just to distinguish them from our Across-specific types.